File: 프로그래머스/lv1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    # 겹치는 숫자 제거 
    lost_2 = list(set(lost) - set(reserve))
    reserve_2 = list(set(reserve) - set(lost))
    
    # lost, reserve 정렬 
    lost_2.sort()
    reserve_2.sort()
    
    # lost_2 안에서 돌면서 
    for i in range(len(lost_2)):
        num = lost_2[i]
        if(len(lost_2) == 0): break
        if (num - 1) in reserve_2: # lost_2 -1 이 
            reserve_2.remove(num - 1)
        elif (num + 1) in reserve_2:
            reserve_2.remove(num + 1)
        else:
            n -= 1
    return n

File: 프로그래머스/lv1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = n;
        
        // 리스트로 만들기 
        List<Integer> list_lost = new ArrayList<>();
        for(int num : lost){
            list_lost.add(num);
        }
        List<Integer> list_reserve = new ArrayList<>();
        for(int num : reserve){
            list_reserve.add(num);
        }
        Collections.sort(list_lost);
        Collections.sort(list_reserve);
        
        List<Integer> diff_lost = new ArrayList<>();
        List<Integer> diff_reserve = new ArrayList<>();
        
        // 중복 제거하기
        diff_lost = list_lost.removeAll(list_reserve);
        diff_reserve = list_reserve.removeAll(list_lost);
        
        for(int num : diff_lost){
            if(diff_lost.size() == 0) break;
            
            if(diff_reserve.contains(num - 1)){
                diff_reserve.remove(diff_reserve.indexOf(num - 1));
            }
            else if (diff_reserve.contains(num + 1)){
                diff_reserve.remove(diff_reserve.indexOf(num + 1));
            }
            else 
            {answer--;}
        }
        return answer;
    }
}